packages/astro/src/transform/babel.ts

```diff
@@ -0,0 +1,30 @@
+import traverse from '@babel/traverse';
+import generator from '@babel/generator';
+import * as t from '@babel/types';
+import parser from '@babel/parser';
+import prettier from 'prettier';
+
+// @ts-ignore @babel/traverse isn't ESM and needs this trick
+export const visit = traverse.default as typeof traverse;
+
+export { t };
+export { default as template } from '@babel/template';
+
+export async function generate(ast: t.File, configURL?: string) {
+	// @ts-ignore @babel/generator isn't ESM and needs this trick
+	const astToText = generator.default as typeof generator;
+
+	const text = astToText(ast, { retainLines: true }).code;
+
+	const prettierOptions = await prettier.resolveConfig(configURL || process.cwd());
+	const formatted = prettier.format(text, {
```

Comment:
how bad is the code change without this? Mind sharing a screenshot? 

The reason is that for every user that appreciates this formatting, there are probably more who don't have a prettier config that this will mess with in a way that will be hard to revert (since it includes the needed code change).

Reply:
also fwiw I just learned that recast preserves original formatting. If we still have rough alignment to migrate to recast long-term, then I'd feel better about this as the short-term fix.

Reply:
It's definitely bad without it, so I'm not sure which option is worse.

<img width="1215" alt="Screen Shot 2022-03-24 at 4 54 37 PM" src="https://user-images.githubusercontent.com/7118177/160016804-b6ea3b9b-4ddb-4837-9cd9-d376fbac6d35.png">

Reply:
those imports! My eyes!

Reply:
Actually, just kidding. It's not as bad as I thought. Comments get a little wonky but there's not much we can do about that.

<img width="1157" alt="Screen Shot 2022-03-24 at 4 59 35 PM" src="https://user-images.githubusercontent.com/7118177/160017390-aac0d504-9a22-4f74-9477-899373cecba4.png">
.

Reply:
oh! that's really not bad at all. strong +1 for that over prettier.